src/extension.js

```diff
@@ -372,17 +289,9 @@ const ServiceIndicator = GObject.registerClass({
 
     async _onServiceChanged(service, pspec) {
```

Comment:
This function should probably be used to set the [`St.Button:checked` property](https://gjs-docs.gnome.org/st10/st.button#property-checked) of the toggle, e.g. something as simple as:

```js
this.checked = service.active;
```

Then add an override to `ServiceToggle` for [`St.Button.clicked()`](https://gjs-docs.gnome.org/st10/st.button#vfunc-clicked):

```js
vfunc_clicked(clickedButton) {
    if (this.service.active)
        this.service.stop().catch(logError);
    else
        this.service.start().catch(logError);
}
```

Thus when the button is clicked it just does the right thing, and the button state changes with the actual service state.

Reply:
Going through with this will cause the extension to not follow the last state on every boot.
Since we won't be updating the enabled key, value registered in gsettings will be considered the default one. For example; if enabled is set to true, it will start the service and enable the toggle regardless of the fact that it may have been toggled off by the user in previous boot. I purpose skipping this change for convenience.

Reply:
So, @andyholmes what should I do?

Reply:
> This function should probably be used to set the [`St.Button:checked` property](https://gjs-docs.gnome.org/st10/st.button#property-checked) of the toggle, e.g. something as simple as:
> 
> ```js
> this.checked = service.active;
> ```
> 
> Then add an override to `ServiceToggle` for [`St.Button.clicked()`](https://gjs-docs.gnome.org/st10/st.button#vfunc-clicked):
> 
> ```js
> vfunc_clicked(clickedButton) {
>     if (this.service.active)
>         this.service.stop().catch(logError);
>     else
>         this.service.start().catch(logError);
> }
> ```
> 
> Thus when the button is clicked it just does the right thing, and the button state changes with the actual service state.

We already have _onEnabledChanged doing vfunc_clicked's job for us. I purpose partially implementing this.